profiles/views.py

```diff
@@ -0,0 +1,48 @@
+from django.shortcuts import render, redirect, reverse, get_object_or_404
+from django.contrib import messages
+
+from .models import UserProfile
+from .forms import UserProfileForm
+
+from checkout.models import Order
+
+
+def profile(request):
+    """A view to return the index page"""
+
+    profile = get_object_or_404(UserProfile, user=request.user)
+
+    if request.method == 'POST':
+        form = UserProfileForm(request.POST, instance=profile)
+        if form.is_valid():
+            form.save()
+            messages.success(request, f'Profile successfully updated')
+            return redirect(reverse('profile'))
+
+        messages.error(request, 'Failed to update profile. Make sure your form is valid')
+        return redirect(reverse('profile'))
+
+    form = UserProfileForm(instance=profile)
+    orders = profile.orders.all()
+
+    template = 'profiles/profile.html'
+    context = {
+        'orders': orders,
+        'form': form
+    }
+
+    return render(request, template, context)
+
+
+def order_history(request, order_number):
```

Comment:
Indeed. Interestingly I believe this exposes a privacy issue as well, because if you knew an order number you could manually type the URL (whether logged in or not) and get the client's full name, phone number and shipping address. In the real world there would definitely need to be some validation that the user requesting the past confirmation is the user that originally ordered the product.